python/GafferSceneUI/_SceneViewInspector.py

```diff
@@ -0,0 +1,959 @@
+					# a spreadsheet due to enabled plug adoption (which passes us onto the default row).
+					if edit.hasEdit and isinstance( node, GafferScene.ShaderAssignment ) :
+						self.__warn( "Edits to '{}' may affect other locations in the scene.", edit.acquireEdit().node() )
+					return edit
+
+				elif edit.hasEdit and editScope is not None :
+					# If an edit scope has been specified, but we're not inside of it, then we shouldn't
+					# use this node. We are downstream of any potential edit though, so do however need
+					# to warn people that any edits they do make may be overridden by this node.
+					displayNode = node.ancestor( Gaffer.EditScope ) or node
+					self.__warn( "Parameter has edits downstream in '{}'.", displayNode )
+
+		# If we haven't found an edit here, walk up the history
+		for p in attributeHistory.predecessors :
+			edit = self.__findEdit( p, editScope )
+			if edit is not None :
+				return edit
+
+		if editScope is not None and not self.__errorMessage:
+			# If we've not been able to find an edit, and we have a scope
+			# specified, then we musn't have passed through the edit scope
+			self.__error( "The target Edit Scope '{}' is not in the scene history.", editScope )
+
+		return None
+
+	def __warn( self, message, node ) :
+
+		self.__warningMessages.append( message.format( self.__displayName( node ) ) )
+
+	def __error( self, message, node ) :
+
+		self.__errorMessage = message.format( self.__displayName( node ) )
+		# We clear out the warnings in the case of an error to prevent confusion in the UI
+		self.__warningMessages = []
+
+	@staticmethod
+	def __displayName( node ) :
+
+		return node.relativeName( node.ancestor( Gaffer.ScriptNode ) )
+
+## \todo Figure out how this relates to the DiffRow in the SceneInspector.
+class _ParameterWidget( GafferUI.Widget ) :
+
+	def __init__( self, parameter ) :
+
+		self.__parameter = parameter
+
+		grid = GafferUI.GridContainer( spacing = 2 )
+		GafferUI.Widget.__init__( self, grid )
+
+		self.__inspectors = []
+
+		with grid :
+
+			GafferUI.Label(
+				## \todo Prettify label text (remove snake case)
+				text = "<h5>" + IECore.CamelCase.toSpaced( parameter ) + "</h5>",
+				parenting = { "index" : ( slice( 0, 2 ), 0 ) }
+			)
+
+			self.__editButton = GafferUI.Button( image = "editOff.png", hasFrame = False,
+				parenting = {
+					"index" : ( 0, 1 ),
+					"alignment" : ( GafferUI.HorizontalAlignment.Center, GafferUI.VerticalAlignment.Center )
+				}
+			)
+			self.__editButton.clickedSignal().connect( Gaffer.WeakMethod( self.__editButtonClicked ), scoped = False )
+
+			self.__valueWidget = _ValueWidget( parenting = { "index" : ( 1, 1 ) } )
+			self.__valueWidget.buttonReleaseSignal().connect( Gaffer.WeakMethod( self.__valueWidgetClicked ), scoped = False )
+
+		self.update( [] )
+
+	def update( self, inspectors ) :
+
+		# The somewhat intricate logic here is attempts to determine whether we should present the user
```

Comment:
here is -> here

Reply:
FIxed [here](https://github.com/GafferHQ/gaffer/pull/3832/commits/3133a70aab9f52308490a9c4e43de4d968e74e64#diff-b9d160a51cf1199c9bca6867338d8f42L659-R663).